src/doc/en/developer/portability_testing.rst

```diff
@@ -60,13 +60,33 @@ Make sure that your Docker client is configured to provide enough RAM
 to the containers (8 GB are a good choice). In Docker Desktop this
 setting is in Preferences -> Resources -> Advanced.
 
+As an alternative, you can also run Docker in GitHub Codespaces
+(or another cloud service) using a container with the Docker-in-Docker
+feature. Sage provides a suitable dev container configuration
+`.devcontainer/tox-docker-in-docker
+<https://github.com/sagemath/sage/tree/develop/.devcontainer/tox-docker-in-docker>`_.
+
+.. only:: html
+
+   .. |codespace-tox-docker-in-docker| image:: https://github.com/codespaces/badge.svg
+      :target: https://codespaces.new/sagemath/sage?devcontainer_path=.devcontainer%2Ftox-docker-in-docker%2Fdevcontainer.json
+
+   .. list-table::
+      :widths: 25 60
+      :header-rows: 0
+
+      * - `tox-docker-in-docker
+          <https://github.com/sagemath/sage/tree/develop/.devcontainer/tox-docker-in-docker>`_
+        - |codespace-tox-docker-in-docker|
+
 All examples in this section were obtained using Docker Desktop for
```

Comment:
Lines 63--80 may be put into `.. note::` as this paragraph breaks the normal flow of narration.

It seems that the list-table is not necessary. The link to the file is already in line 66--67. How about just displaying the codespaces button (this button would be part of the `.. note::`)?

Reply:
I am happy with this PR except this issue.

Reply:
Like this
```
Make sure that your Docker client is configured to provide enough RAM
to the containers (8 GB are a good choice). In Docker Desktop this
setting is in Preferences -> Resources -> Advanced.

.. NOTE::

   All examples in this section were obtained using Docker Desktop for
   Mac; but the `command-line user interface
   <https://docs.docker.com/engine/reference/commandline/cli/>`_ for the
   other platforms is identical.

   As an alternative, you can also run Docker in GitHub Codespaces
   (or another cloud service) using a container with the Docker-in-Docker
   feature. Sage provides a suitable dev container configuration
   `.devcontainer/tox-docker-in-docker
   <https://github.com/sagemath/sage/tree/develop/.devcontainer/tox-docker-in-docker>`_:

   .. ONLY:: html

      .. image:: https://github.com/codespaces/badge.svg
         :align: center
         :target: https://codespaces.new/sagemath/sage?devcontainer_path=.devcontainer%2Ftox-docker-in-docker%2Fdevcontainer.json

All major Linux distributions provide ready-to-use Docker images,   
```

Reply:
![Screen Shot 2024-01-11 at 1 59 11 PM](https://github.com/sagemath/sage/assets/1456101/9cfd206f-8e2b-42d0-8578-bc2acf74f49b)

Reply:
Thanks, that's a nice improvement. I've committed it as 390c949db7b